jruby-gradle-base-plugin/src/main/groovy/com/github/jrubygradle/GemUtils.groovy

```diff
@@ -156,6 +156,70 @@ class GemUtils {
         extractGems(project,jRubyClasspath,project.files(gemConfig.files),destDir,action)
     }
 
+    static void setupJars(Configuration config,
+                          File destDir,
+                          GemUtils.OverwriteAction overwrite) {
+        File jarsDir = new File(destDir, 'jars')
+        def artifacts = config.resolvedConfiguration.resolvedArtifacts
+        def fileRenameMap = [:]
+        def coordinates = []
+        def files = []
+        artifacts.each { dep ->
+
+            def group = dep.moduleVersion.id.group
+            def groupAsPath = group.replace('.' as char, File.separatorChar)
+            def version = dep.moduleVersion.id.version
+            // TODO classifier
+            def newFileName = "${groupAsPath}/${dep.name}/${version}/${dep.name}-${version}.${dep.type}"
+
+            if (group != 'rubygems' && dep.type != 'gem' && dep.name != 'jruby-complete') {
+                // TODO classifier and system-scope
+                coordinates << "${group}:${dep.name}:${version}:runtime:"
+            }
+            fileRenameMap[dep.file.name] = newFileName
+            // TODO omit system-scoped files
+            files << dep.file
+        }
+
+        // create Jars.lock file used by jar-dependencies
+        File jarsLock = new File(destDir, 'Jars.lock')
+        switch(overwrite) {
+            case OverwriteAction.FAIL:
+                if (jarsLock.exists()) {
+                    throw new DuplicateFileCopyingException("Jars.lock already exists")
+                }
+            case OverwriteAction.SKIP:
+                if (jarsLock.exists()) {
+                  break
+                }
+            case OverwriteAction.OVERWRITE:
+                jarsLock.parentFile.mkdirs()
+                jarsLock.withWriter { writer ->
+                    coordinates.each { writer.println it }
+                }
+        }
+
+        files.each { File f ->
```

Comment:
Similar to my comment on the above block, I think this block (L202-L220) can/should be broken out into an independent method so it can be better tested, e.g.

```groovy
/**
 * @return boolean true if we could successfully rewrite the dependency's file
 * @throws DuplicateFileCopyingException failure to overwrite a dependency file which already exists
 */
static boolean rewriteDependencyFiles(List<File> dependencies, Map<String, String> renameMap)
```

The unit test for this method I think we be similarly easy to verify that our `overwrite` behavior is correct